service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -149,8 +150,16 @@ private String buildLotNumberPertinentInformation(Immunization immunization) {
     }
 
     private String buildExpirationDatePertinentInformation(Immunization immunization) {
-        Optional<Date> expirationDate = Optional.ofNullable(immunization.getExpirationDate());
-        return expirationDate.map(value -> EXPIRATION + formatShortDate(value)).orElse(StringUtils.EMPTY);
+        Optional<DateType> expirationDateElement = Optional.ofNullable(immunization.getExpirationDateElement());
+        if (expirationDateElement.isPresent()) {
+            if (expirationDateElement.get().hasValue()) {
+                return expirationDateElement.map(dateType -> EXPIRATION + formatDateType(dateType)).orElse(StringUtils.EMPTY);
+            } else {
+                return StringUtils.EMPTY;
+            }
+        } else {
+            return StringUtils.EMPTY;
+        }
```

Comment:
```suggestion
        Optional<DateType> expirationDateElement = Optional.ofNullable(immunization.getExpirationDateElement());
        if (expirationDateElement.isPresent() && expirationDateElement.get().hasValue()) {
            return expirationDateElement.map(dateType -> EXPIRATION + formatDateType(dateType)).orElse(StringUtils.EMPTY);
        } 
           
        return StringUtils.EMPTY;
```

Reply:
There could be only one if and one StringUtils.EMPTY; to achieve the same effect.